File: app/src/main/java/com/jeksonshar/jweather/repository/room/Converter.java
package com.jeksonshar.jweather.repository.room;

import com.jeksonshar.jweather.model.WeatherModel;

import java.util.Date;

public class Converter {

    static WeatherEntity convert(WeatherModel weatherModel) {

        WeatherEntity weatherEntity = new WeatherEntity();

        weatherEntity.id = String.valueOf(weatherModel.getId());
        weatherEntity.date = weatherModel.getDate().getTime();
        weatherEntity.weatherStateAbbr = weatherModel.getWeatherStateAbbr();
        weatherEntity.weatherState = weatherModel.getWeatherState();
        weatherEntity.tempMax = weatherModel.getTempMax();
        weatherEntity.tempMin = weatherModel.getTempMin();
        weatherEntity.windSpeed = weatherModel.getWindSpeed();
        weatherEntity.airPressure = weatherModel.getAirPressure();
        weatherEntity.humidity = weatherModel.getHumidity();

        return weatherEntity;
    }

    static WeatherModel convert(WeatherEntity weatherEntity) {

        WeatherModel weatherModel = null;

        if (weatherEntity != null) {

            weatherModel = new WeatherModel();

            weatherModel.setId(Long.parseLong(weatherEntity.id));

            Date date = new Date();
            date.setTime(weatherEntity.date);
            weatherModel.setDate(date);

            weatherModel.setWeatherStateURL(weatherEntity.weatherStateAbbr);
            weatherModel.setWeatherState(weatherEntity.weatherState);
            weatherModel.setTempMax(weatherEntity.tempMax);
            weatherModel.setTempMin(weatherEntity.tempMin);
            weatherModel.setWindSpeed(weatherEntity.windSpeed);
            weatherModel.setAirPressure(weatherEntity.airPressure);
            weatherModel.setHumidity(weatherEntity.humidity);
        }
        return weatherModel;
    }
}

File: app/src/main/java/com/jeksonshar/jweather/repository/room/Converter.kt
package com.jeksonshar.jweather.repository.room

import com.jeksonshar.jweather.model.WeatherModel
import java.util.*

object Converter {
    fun convert(weatherModel: WeatherModel): WeatherEntity {

        val weatherEntity = WeatherEntity()

        weatherEntity.id = weatherModel.id.toString()
        weatherEntity.date = weatherModel.date!!.time
        weatherEntity.weatherStateAbbr = weatherModel.weatherStateAbbr!!
        weatherEntity.weatherState = weatherModel.weatherState!!
        weatherEntity.tempMax = weatherModel.tempMax
        weatherEntity.tempMin = weatherModel.tempMin
        weatherEntity.windSpeed = weatherModel.windSpeed
        weatherEntity.airPressure = weatherModel.airPressure
        weatherEntity.humidity = weatherModel.humidity

        return weatherEntity
    }

    fun convert(weatherEntity: WeatherEntity?): WeatherModel? {

        var weatherModel: WeatherModel? = null

        if (weatherEntity != null) {

            weatherModel = WeatherModel()

            weatherModel.id = weatherEntity.id.toLong()

            val date = Date()
            date.time = weatherEntity.date
            weatherModel.date = date

            weatherModel.setWeatherStateURL(weatherEntity.weatherStateAbbr)
            weatherModel.weatherState = weatherEntity.weatherState
            weatherModel.tempMax = weatherEntity.tempMax
            weatherModel.tempMin = weatherEntity.tempMin
            weatherModel.windSpeed = weatherEntity.windSpeed
            weatherModel.airPressure = weatherEntity.airPressure
            weatherModel.humidity = weatherEntity.humidity
        }
        return weatherModel
    }
}